R/01-qc.R

```diff
@@ -1,119 +1,50 @@
-
 #' A function to run QC
 #' @description This is a function here's where we describe what it does
-#' @param plasmid_cutoff default it NULL; `qc_plasmid_histogram` will calculate and set a default one unless a value is specified here 
+#' @param plasmid_cutoff default it NULL; `qc_plasmid_histogram` will calculate and set a default one unless a value is specified here
 #' @param use_combined default it TRUE; if TRUE, both zero count and low plasmid CPM filters are applied and if either is TRUE, a pgRNA construct will be filtered out. If FALSE, need to allow to specify which should be used
-#' @param params default is `list(cell_line = NULL)`; used to pass info when saving plots
 #' @param plots_dir default is `./qc_plots`; directory to save plots created with this function, if it doesn't exist already it will be created
 #' @param overwrite default is FALSE; whether to overwrite the QC Report file
-#' @param output_format default is NULL; the output format that the QC Rmd will be knitted to/produced
-#' @param output_file default is `QC_Report`; name of the output QC report file
-#' @param output_dir default is `./`; where the qc report will be stored
-#' @param wide_ar default is 0.75; aspect ratio
+#' @param output_file_path default is `QC_Report`; name of the output QC report file
+#' @param ...
 #' @export
 #' @importFrom tidyr pivot_longer
 #' @importFrom magrittr %>%
 #' @examples \dontrun{
 #'
 #' }
+run_qc <- function(gimap_dataset,
+                   plots_dir = "./qc_plots",
+                   overwrite = FALSE,
+                   output_format = NULL,
+                   output_file = "QC_Report",
+                   output_dir = "./",
+                   plot_suffix = "png") {
 
-
-run_qc <- function(gimap_dataset, plasmid_cutoff = NULL, use_combined = TRUE,
-                   params = list(cell_line = NULL), plots_dir = "./qc_plots", overwrite = FALSE, output_format = NULL, output_file = "QC_Report", output_dir = "./",
-                   wide_ar = 0.75) {
-  
-  store_results <- function(result_obj, lvoi, list_of_qc_things, plots_dir, params){
```

Comment:
If we don't need to store the results, which is the strategy I like, then we can remove this function -- one less thing to maintain. But let me know your thoughts.

Reply:
Agreed I only added this function because I was storing the plots as well as the locations where the plots were stored and was trying to reduce receptiveness.